app/javascript/mastodon/features/notifications/index.js

```diff
@@ -20,19 +20,36 @@ const messages = defineMessages({
   title: { id: 'column.notifications', defaultMessage: 'Notifications' },
 });
 
+const collapsibleNotifications = (a, b) => {
+  if (!['reblog', 'favourite'].includes(a.get('type'))) {
+    return false;
+  }
+
+  return a.get('type') === b.get('type') && a.get('status') === b.get('status');
```

Comment:
Works, although now, a single toot being alternatively boosted and favourited (which is very frequent for trending toots) will prevent collapsing.